go/test/utils/rows_test.go

```diff
@@ -0,0 +1,172 @@
+	"[[INT64(2) VARCHAR(\"Article 2\") INT64(10)]]",
+	"[[INT64(1) VARCHAR(\"Article 1\") INT64(10)]]",
+	"[[INT64(2) VARCHAR(\"Article 2\") INT64(10)]]",
+	"[[VARCHAR(\"albumQ\") INT32(4)] [VARCHAR(\"albumY\") INT32(1)] [VARCHAR(\"albumY\") INT32(2)] [VARCHAR(\"albumX\") INT32(2)] [VARCHAR(\"albumX\") INT32(3)] [VARCHAR(\"albumX\") INT32(1)]]",
+	"[[VARCHAR(\"albumQ\") INT32(4)] [VARCHAR(\"albumY\") INT32(1)] [VARCHAR(\"albumY\") INT32(2)] [VARCHAR(\"albumX\") INT32(2)] [VARCHAR(\"albumX\") INT32(3)] [VARCHAR(\"albumX\") INT32(1)]]",
+	"[[INT64(2)]]",
+	"[[INT32(1) INT32(100)]]",
+	"[[INT32(1) INT32(100)]]",
+	"[[INT64(2)]]",
+	"[[INT64(2)]]",
+	"[[INT64(2)]]",
+	"[[INT64(2)]]",
+	"[[UINT32(70)]]",
+	"[[INT64(1) VARCHAR(\"Article 1\") INT64(20)]]",
+	"[[INT64(2) VARCHAR(\"Article 2\") INT64(20)]]",
+	"[[INT64(1) VARCHAR(\"Article 1\") INT64(20)]]",
+	"[[INT64(2) VARCHAR(\"Article 2\") INT64(20)]]",
+	"[[INT64(1) VARCHAR(\"Article 1\") INT64(10)]]",
+	"[[INT64(2) VARCHAR(\"Article 2\") INT64(10)]]",
+	"[]",
+	"[]",
+	"[[INT64(1) NULL] [INT64(2) INT64(2)]]",
+	"[[INT64(1) INT64(1)] [INT64(2) NULL]]",
+	"[[INT64(1) INT64(1)]]",
+	"[[INT64(1) INT64(8)] [INT64(1) INT64(9)]]",
+	"[[INT64(1)] [INT64(2)]]",
+	"[[INT64(1)]]",
+	"[[INT64(4)] [INT64(8)] [INT64(12)]]",
+	"[[INT64(1)]]",
+	"[[INT64(1)]]",
+	"[[INT64(1)]]",
+	"[]",
+	"[]",
+	"[[INT64(1)]]",
+	"[[INT64(1)]]",
+	"[[INT64(1)]]",
+	"[]",
+	"[]",
+	"[[INT64(1)]]",
+	"[[INT64(1)]]",
+	"[[DECIMAL(2) INT64(1)]]",
+	"[[NULL INT64(0)]]",
+	"[[DECIMAL(420) INT64(1)]]",
+	"[[DECIMAL(420) INT64(1)]]",
+	"[[NULL INT64(0)]]",
+	"[]",
+	"[[NULL INT64(0)]]",
+	"[]",
+	"[[DECIMAL(3) INT64(3)]]",
+	"[[DECIMAL(2) INT64(1)] [DECIMAL(1) INT64(1)] [DECIMAL(0) INT64(1)]]",
+	"[[NULL INT64(0)]]",
+	"[]",
+	"[[DECIMAL(423) INT64(4)]]",
+	"[[DECIMAL(423) INT64(4)]]",
+	"[[DECIMAL(420) INT64(1)] [DECIMAL(2) INT64(1)] [DECIMAL(1) INT64(1)] [DECIMAL(0) INT64(1)]]",
+	"[[DECIMAL(420) INT64(1)]]",
+	"[[DECIMAL(420) INT64(1)]]",
+	"[[INT64(1) INT64(2)]]",
+	"[[INT64(1) INT64(2)] [INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(2)]]",
+	"[[INT64(1) INT64(2)] [INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(2)]]",
+	"[[INT64(1) INT64(2)] [INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(2)]]",
+	"[[INT64(1) INT64(2)] [INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(2)]]",
+	"[[INT64(1) INT64(2)] [INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(4)]]",
+	"[[INT64(1) INT64(2)]]",
+	"[[INT64(1) INT64(2)]]",
+	"[[INT64(2) INT64(4)]]",
+	"[[INT64(2) INT64(4)]]",
+	"[[INT64(1) INT64(1)] [INT64(1) INT64(2)] [INT64(1) INT64(3)]]",
+	"[[INT64(1) INT64(1)] [INT64(1) INT64(2)] [INT64(1) INT64(3)] [INT64(1) INT64(4)] [INT64(1) INT64(5)] [INT64(1) INT64(6)]]",
+	"[[INT64(1) INT64(1)] [INT64(1) INT64(2)] [INT64(1) INT64(3)]]",
+}
+
+func TestRowParsing(t *testing.T) {
+	for _, r := range TestRows {
+		output, err := ParseRows(r)
+		if err != nil {
```

Comment:
if only we had some test library that allowed us to write `NoError(t, err)`

/s